sympy/solvers/ode/systems.py

```diff
@@ -0,0 +1,375 @@
+        'order': order,
+        'is_linear': is_linear,
+        'is_constant': is_constant,
+        'is_homogeneous': is_homogeneous,
+    }
+
+    Dict or None
+        Dict with values for keys:
+            1. no_of_equation: Number of equations
+            2. eq: The set of equations
+            3. func: List of dependent variables
+            4. order: A dictionary that gives the order of the
+                      dependent variable in eqs
+            5. is_linear: Boolean value indicating if the set of
+                          equations are linear or not.
+            6. is_constant: Boolean value indicating if the set of
+                          equations have constant coefficients or not.
+            7. is_homogeneous: Boolean value indicating if the set of
+                          equations are homogeneous or not.
+        This Dict is the answer returned if the eqs are linear and constant
+        coefficient. Otherwise, None is returned.
+    """
+    from sympy.solvers.solveset import linear_eq_to_matrix
+    from sympy.solvers.solvers import solve
+
+    # Error for i == 0 can be added but isn't for now
+
+    # Removing the duplicates from the list of funcs
+    # meanwhile maintaining the order. This is done
+    # since the line in classify_sysode: list(set(funcs)
+    # cause some test cases to fail when gives different
+    # results in different versions of Python.
+    funcs = list(uniq(funcs))
+
+    # Check for len(funcs) == len(eqs)
+    if len(funcs) != len(eqs):
+        raise ValueError("Number of functions given is not equal to the number of equations %s" % funcs)
+
+    # ValueError when functions have more than one arguments
+    for func in funcs:
+        if len(func.args) != 1:
+            raise ValueError("dsolve() and classify_sysode() work with "
+            "functions of one variable only, not %s" % func)
+
+    # Getting the func_dict and order using the helper
+    # function
+    order = _get_func_order(eqs, funcs)
+
+    # Not adding the check if the len(func.args) for
+    # every func in funcs is 1
+
+    rep = {func.diff(t, n): Dummy() for func in funcs for n in range(order[func] + 1)}
+    eqs_sub = [eq.subs(rep) for eq in eqs]
+
+    # Linearity check
+    try:
+        A, b = linear_eq_to_matrix(eqs_sub, rep.values())
+
+    except ValueError:
+        return None
+
+    is_linear = True
+
+    # Constant coefficient check
+    is_constant = True
+    for coef in A:
+        if is_constant == True:
+            if coef.as_independent(t, as_Add=True)[1] != 0:
+                is_constant = False
+
+    # Homogeneous check
+    is_homogeneous = True if b.is_zero_matrix else False
+
+    match = {
+        'no_of_equation': len(eqs),
+        'eq': eqs,
+        'func': funcs,
+        'order': order,
+        'is_linear': is_linear,
+        'is_constant': is_constant,
+        'is_homogeneous': is_homogeneous,
+    }
+
+    if match['is_linear'] and match['is_constant']:
```

Comment:
Here `is_linear` is always True.

Reply:
@oscarbenjamin That is true but in future when this function is updated, it may not always be ```True```. I thought of that and didn't remove this currently unnecessary condition. But I will update it for now and add a comment describing that this may have to be updated.

Reply:
Okay, fair enough. I guess this can be simplified later when the other solvers are added.